### PR TITLE
Add avatar dav endpoint

### PR DESCRIPTION
**Pull request by rullzer**
This add an avatar collection to the new dav endpoint. Right now it is still authenticated but once we figure it out properly we can have anonymous actions here (for getting!) avatars. But for now it will allow authenticated clients to retrieve avatars for local users.

Pretty straight forward actually.

CC: @DeepDiver1975 @PVince81 @nickvergessen

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Tested and works nicely :) :+1:

**Comment by DeepDiver1975** [comment]
@rullzer please add the extension to the url.

see https://github.com/DeepDiver1975/dav/blob/master/lib/avatar/avatarhome.php#L52-L65 :wink:

**Comment by DeepDiver1975** [comment]
a request to get the avatar should really look like

````
GET http://server/remote.php/dav/avatars/$user/128.png

**Comment by rullzer** [comment]
@DeepDiver1975 no. Since avatars can be jpg or png. There is no way to know which beforehand (or we need to convert them on the fly). The old endpoint also does not have this and I assume it is for the same reason.

**Comment by PVince81** [comment]
If users were Webdav resources, they could all have a "avatar" property that points to the URL of the avatar with the correct extension.

**Comment by rullzer** [comment]
I don't care how we'll handle it. I just want to make sure that there is only 1 request to the server to obtain avatars :P And not 2.

**Comment by DeepDiver1975** [comment]
> There is no way to know which beforehand (or we need to convert them on the fly).

we better do - see my code - thx

**Comment by DeepDiver1975** [comment]
> If users were Webdav resources, they could all have a "avatar" property that points to the URL of the avatar with the correct extension.

a property like that can be added easily - caldav and carddav define a home propery - see http://icalendar.org/CalDAV-Access-RFC-4791/6-2-1-caldav-calendar-home-set-property.html

**Comment by rullzer** [comment]
Yes I was planning to do that in a followup PR. But still this adds an extra server call for all avatars

**Comment by rullzer** [comment]
Ok so as discussed on IRC:

Probabaly it makes sense to add the availibity of an avatar to the shareeAPI. That way clients will not retrieve the avatar if it is not there.

If the avatar is available we could include an URL template. Something like

```
avatar = {
  'available' : true,
  'baseurl' : 'https://server/remote.php/dav/avatars/userid/
  'ext' : 'png'
}
```

That way clients can themself efficiently generate the URL.

So... TODO:

- [ ] Update Sharee API
- [ ] Update URL generator
- [ ] Update dav avatar endpoint with ext

**Comment by rullzer** [comment]
Ok now with file extension.

I'll fix the other stuff in other PR's since it different parts of the code and code wise unrelated.

File: apps/dav/lib/avatars/avatar.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
+<?php
+
```
Comment (nickvergessen) [inline apps/dav/lib/avatars/avatar.php]: No license :no_good:

File: apps/dav/lib/avatars/avatar.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	/**
+	 * Get the file size of the avatar
+	 *
+	 * @return int
+	 */
+	public function getSize() {
+		return $this->avatar->getSize();
+	}
+
+}
```
Comment (LukasReschke) [inline apps/dav/lib/avatars/avatar.php]: New line at EoF

**Comment by nickvergessen** [comment]
> Probabaly it makes sense to add the availibity of an avatar to the shareeAPI. That way clients will not retrieve the avatar if it is not there.

Does this mean you will then in the api call all the remote web davs to check the availability? makes no sense to me, or did I not understand you correctly?

**Comment by rullzer** [comment]
@nickvergessen no that is not what I ment.

The whole story is that I talked to @DeepDiver1975 and when we have carddav sharing for federated shares (so your ownCloud instance 'knows' the users on my instance). That those vcards also contain the avatar. Thus if we know the remote we could easily generate the avatar locally eventually.

It is something we need to think more about to streamline (Where to store those avatars... etc). But the idea seems nicer.

**Comment by rullzer** [comment]
Let me close this as it will need to be reworked largely anyway.

**Pull request closed without merging**